cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -93,6 +93,14 @@
   void dropPartition(PartitionKey key);
 
   /**
+   * Concatenates the partition in Hive. If this Dataset is not enabled for Explore, this operation is a no-op.
```

Comment:
Also, I wonder if we should suggest in the javadoc to perform this operation outside of a transaction, as it seems fairly likely that this will take longer than 30 seconds.